spec/dynamoid/dumping_spec.rb

```diff
@@ -366,6 +368,18 @@
         expect(raw_attributes(obj)[:signed_up_on]).to eql('2017-09-25')
       end
 
+      it 'stores in string format with precision when :iso_precision is present' do
+        klass = new_class do
+          field :signed_up_on, :datetime, store_as_string: true, iso_precision: 6
+        end
+
+        signed_up_on = DateTime.now.utc.iso8601(6)
+
+        obj = klass.create(signed_up_on: signed_up_on)
+
+        expect(reload(obj).signed_up_on.to_f).to eql(DateTime.iso8601(signed_up_on).to_f)
```

Comment:
I think i't completely OK to use `to_f` here but it's a bad practice to check equality of floating point numbers. Compare milliseconds or string representation would be more correct.

The precision is 6 in the test so it's OK, but with precision 9 (with nanoseconds) this test could become flaky because MRI isn't accurate in preserving nanoseconds for Time/DateTime.